notebooks/NIRISS_WFSS_advanced/01_niriss_wfss_image2_image3.ipynb

```diff
@@ -0,0 +1,1070 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d1addba0-ae56-4565-b12a-3ded03d9b0bf",
+   "metadata": {},
+   "source": [
+    "#### ImViz\n",
+    "\n",
+    "Similarly to DS9, ImViz allows you to interactively view these images and the corresponding source catalog as well."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7bec8a55",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imviz = Imviz()\n",
+    "viewer = imviz.default_viewer\n",
+    "\n",
+    "for img in image3_i2d:\n",
+    "    print(f'Plotting: {img}')\n",
+    "    label = f\"obs{fits.getval(img, 'OBSERVTN')} {fits.getval(img, 'PUPIL')}\"\n",
+    "    with warnings.catch_warnings():\n",
+    "        warnings.simplefilter('ignore')\n",
+    "        imviz.load_data(img, data_label=label)\n",
+    "\n",
+    "    # this aligns the image to use the WCS coordinates\n",
+    "    linking = imviz.plugins['Links Control']\n",
+    "    linking.link_type = 'WCS'\n",
+    "\n",
+    "    # also plot the associated catalog\n",
+    "    cat = Table.read(img.replace('i2d.fits', 'cat.ecsv'))\n",
+    "    # format the table into the format imviz expects\n",
+    "    t_xy = Table({'x': cat['xcentroid'],\n",
+    "                  'y': cat['ycentroid']})\n",
+    "    viewer.marker = {'color': 'orange', 'alpha': 1, 'markersize': 20, 'fill': False}\n",
+    "    viewer.add_markers(t_xy, marker_name=f\"{label} catalog\")\n",
+    "\n",
+    "# This changes the stretch of all of the images\n",
+    "plotopt = imviz.plugins['Plot Options']\n",
+    "plotopt.select_all(viewers=True, layers=True)\n",
+    "plotopt.stretch_preset = '99.5%'\n",
+    "    \n",
+    "imviz.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ea23a1b-4b6f-4ee1-9e2b-ff132dd9127d",
+   "metadata": {},
+   "source": [
+    "<a id='custom'></a>\n",
+    "## Custom Imaging Pipeline Run"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14216c26-cf89-4f15-b7ec-b5a8ca0071ba",
+   "metadata": {},
+   "source": [
+    "<a id='custom_image3'></a>\n",
+    "### Image3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ccc697e5-92de-4a2a-a9e1-3f81d20fcf27",
+   "metadata": {},
+   "source": [
+    "Try editing a few parameters and compare the outcomes to the default run above, at first for a single file.\n",
+    "\n",
+    "When we call the image3 pipeline, we can add modifications to a specific step of the pipeline. In this case we're going to edit the `source_catalog` and `tweakreg` steps of the pipeline. An explanation of the different parameters to tweak can be found in the further information below, while the default values are a combination of both the default pipeline values listed in there, and the parameter reference files that are supplied.\n",
+    "- [source_catalog Further Information](https://jwst-pipeline.readthedocs.io/en/latest/api/jwst.source_catalog.SourceCatalogStep.html)\n",
```

Comment:
Unless somebody else on the NIRISS team has suggestions, I think it is ok to just link to the docs here. I do not remember why these specific values were chosen.